main.cpp

```diff
@@ -126,7 +136,138 @@ i cubed: 531441
 Use a service like https://www.diffchecker.com/diff to compare your output. 
 */
 
-#include <iostream>
+
+template<class NumType>
+struct Numeric
+{
+    //using Type = Temporary<NumType>;
+    using Type = NumType;
+
+    Numeric(Type nuNum) : value( std::make_unique<Type>(nuNum) ) { } 
+
+    operator Type() const {return *value;} //possibly this line
+
+    template<class MType>
+    Numeric& operator+=( const MType& rhs )
+    {
+        *value += static_cast<Type>(rhs);
+        return *this;
+    }
+    template<class MType>
+    Numeric& operator-=( const MType& rhs )
+    {
+        *value -= static_cast<Type>(rhs);
+        return *this;
+    }
+    template<class MType>
+    Numeric& operator*=( const MType& rhs )
+    {
+        *value *= static_cast<Type>(rhs);
+        return *this;
+    }
+
+    template<class MType>
+    Numeric& operator/=( const MType& rhs )
+    {
+        if constexpr ( std::is_same< Type, int >::value )
+        { 
+            if constexpr ( std::is_same< MType, int >::value )
+            {
+                if ( rhs == 0 )
+                {
+                    std::cout << "error: integer division by zero is an error and will crash the program!" << std::endl;
+                    return *this;
+                }
+            }
+            else if ( std::abs(rhs) < std::numeric_limits<MType>::epsilon() )
+            {
+                std::cout << "can't divide integers by zero!" << std::endl;
+                return *this;
+            }
+        }
+        else if ( std::abs(rhs) < std::numeric_limits<Type>::epsilon() )
+        {
+            std::cout << "warning: floating point division by zero!" << std::endl;
+        }
+        *value /= rhs;
+        return *this;
+    }
+    
+    template<class MType>
+    Numeric& pow(MType& expo)
+    {
+        *value = static_cast<Type>(std::pow( *value, static_cast<Type>(expo) ) );
+        return *this;
+    }
+
+    template<typename Callable>
+    Numeric& apply( Callable&& func )
+    {
+        func(value);
+        return *this;
+    }
+
+    template<typename MType>
+    Numeric& operator=(const MType& rhs)
+    {
+        *value = static_cast<Type>(rhs);
+        return *this;
+    }
+
+    // (const objects can only call their const member functions)
+    // operator Type() const { return *value; }
+    // operator Type&() { return *value; }
+    operator Numeric() const { /* read-only function */ return *value; }
+    operator Numeric&() { /* read/write function */ return *value; }
+
```

Comment:
Your class is `Numeric<>`.  
These two functions just try to convert whatever your `value` owns into another `Numeric` instance.
That's not going to work because you don't have a constructor in this `Numeric` class which takes a `Temporary<>` as its single parameter.
- [ ] delete these conversion functions.